src/script/script_gui.cpp

```diff
@@ -799,25 +800,22 @@ struct ScriptDebugWindow : public Window {
 		this->DrawWidgets();
 	}
 
-	void SetStringParameters(WidgetID widget) const override
+	std::string GetWidgetString(WidgetID widget, StringID stringid) const override
 	{
-		if (widget != WID_SCRD_NAME_TEXT) return;
+		if (widget != WID_SCRD_NAME_TEXT) return this->Window::GetWidgetString(widget, stringid);
 
 		if (this->filter.script_debug_company == OWNER_DEITY) {
 			const GameInfo *info = Game::GetInfo();
 			assert(info != nullptr);
-			SetDParam(0, STR_AI_DEBUG_NAME_AND_VERSION);
-			SetDParamStr(1, info->GetName());
-			SetDParam(2, info->GetVersion());
-		} else if (this->filter.script_debug_company == CompanyID::Invalid() || !Company::IsValidAiID(this->filter.script_debug_company)) {
-			SetDParam(0, STR_EMPTY);
-		} else {
-			const AIInfo *info = Company::Get(this->filter.script_debug_company)->ai_info;
-			assert(info != nullptr);
-			SetDParam(0, STR_AI_DEBUG_NAME_AND_VERSION);
-			SetDParamStr(1, info->GetName());
-			SetDParam(2, info->GetVersion());
+			return GetString(STR_AI_DEBUG_NAME_AND_VERSION, info->GetName(), info->GetVersion());
 		}
+		if (this->filter.script_debug_company == CompanyID::Invalid() || !Company::IsValidAiID(this->filter.script_debug_company)) {
+			return {};
+		}
+
+		const AIInfo *info = Company::GetIfValid(this->filter.script_debug_company)->ai_info;
+		assert(info != nullptr);
+		return GetString(STR_AI_DEBUG_NAME_AND_VERSION, info->GetName(), info->GetVersion());
```

Comment:
Not your work, but this is hard to ignore...

I usually like early returns, especially in void functions, but this definitely feels like it should end with `default: return this->Window::GetWidgetString(widget, stringid);` or at least a final return after a bunch of `if` statements, like the other overrides of `GetWidgetString()` that I've reviewed. (This is done a few other times in this file, but not as oddly.)

Up to you whether you want to standardize this file here or if that's out of scope. 🙂